library/BrazeEngineUserJourneyHandler.m

```diff
@@ -0,0 +1,167 @@
+NSString *START_TIMESTAMP_KEY = @"start_timestamp";
+NSString *END_TIMESTAM_KEY = @"end_timestamp";
+NSString *DURATION_KEY = @"duration";
+
+NSString *IS_HOME_KEY = @"is_home";
+NSString *IS_WORK_KEY = @"is_work";
+NSString *INGRESS_LATITUDE_KEY = @"ingress_latitude";
+NSString *INGRESS_LONGITUDE_KEY = @"ingress_longitude";
+
+NSString *COUNTRY_KEY = @"country";
+NSString *LOCALITIES_KEY = @"localities";
+NSString *POSTCODE_KEY = @"postcode";
+NSString *REGION_KEY = @"region";
+
+NSString *SPAN_ATTACHED_PLACE_EVENT_KEY = @"engine_span_attached_place";
+NSString *ATTACHED_PLACE_NAME_KEY = @"name";
+NSString *ATTACHED_PLACE_ID_KEY = @"factual_id";
+NSString *ATTACHED_PLACE_LATITUDE_KEY = @"latitude";
+NSString *ATTACHED_PLACE_LONGITUDE_KEY = @"longitude";
+NSString *ATTACHED_PLACE_CATEGORIES_KEY = @"category_labels";
+NSString *ATTACHED_PLACE_CHAIN_KEY = @"chain_name";
+NSString *ATTACHED_PLACE_DISTANCE_KEY = @"distance";
+NSString *ATTACHED_PLACE_LOCALITY_KEY = @"locality";
+
+- (id) initWithMaxAttachedPlaceEventsPerEvent:(int)maxAttachedPlaceEventsPerEvent {
+  if (self = [super init]) {
+    _maxAttachedPlacesPerEvent = maxAttachedPlaceEventsPerEvent;
+  }
+  
+  return self;
+}
+
+- (id) init {
+  return [self initWithMaxAttachedPlaceEventsPerEvent:20];
+}
+
+- (void)userJourneyEventDidOccur:(nonnull UserJourneyEvent *)userJourneyEvent { /* Not supported */ }
+
+- (void)userJourneySpanDidOccur:(nonnull UserJourneySpan *)userJourneySpan {
+  // Only send span data packet when did travel is false and tracking User Journey Spans is enabled
+  if (![BrazeEngine isTrackingSpans] || userJourneySpan.didTravel) {
+    return;
+  }
+  
+  // Span information
+  NSString *spanId = userJourneySpan.spanId;
+  BOOL startTimestampUnavailable = userJourneySpan.startTimestampUnavailable;
+  BOOL endTimestampUnavailable = userJourneySpan.endTimestampUnavailable;
+  UInt64 startTimestamp = userJourneySpan.startTimestamp.timeIntervalSince1970 * 1000;
+  UInt64 endTimestamp = userJourneySpan.endTimestamp.timeIntervalSince1970 * 1000;
+  
+  // Current place information
+  FactualPlaceVisit *currentPlace = userJourneySpan.currentPlace;
+  NSNumber *isHome = [[NSNumber alloc] initWithBool:currentPlace.isHome];
+  NSNumber *isWork = [[NSNumber alloc] initWithBool:currentPlace.isWork];
+  
+  // Ingress information
+  CLLocation *ingressLocation = currentPlace.ingressLocation;
+  NSNumber *ingressLatitude = @(ingressLocation.coordinate.latitude);
+  NSNumber *ingressLongitude = @(ingressLocation.coordinate.longitude);
+  
+  // Geographies information
+  Geographies *geographies = currentPlace.geographies;
+  NSString *country = geographies.country;
+  NSArray<NSString*> *localities = geographies.localities;
+  NSString *postcode = geographies.postcode;
+  NSString *region = geographies.region;
+  
+  // Get duration of span
+  NSNumber *duration = !startTimestampUnavailable && !endTimestampUnavailable ? @(endTimestamp - startTimestamp) : 0;
```

Comment:
The `0` here is effectively declaring a `nil` `NSNumber` -- but I'm assuming what we really want is the numeric value zero, `@(0)`.